### ISSUE TITLE
Blue/Green rollout fails due to collisions

### ISSUE THREAD

**Issue by johnkost** [argoproj/argo-rollouts#505]
Occasionally during a blue/green rollout, my rollout fails to progress after the first run of the  `postPromotionAnalysis`. Scrolling through the `rollouts` controller logs, I see the following:
```
time="2020-05-13T19:45:52Z" level=info msg="Encountered collision of existing analysisrun os-demo-web-v2020-04-21-391b-5bc768f98-48 (phase: Successful, equal: false, controllerUIDEqual: true)" namespace=os-demo-jkost rollout=os-demo-web-v2020-04-21-391b
  | time="2020-05-13T19:45:53Z" level=info msg="Encountered collision of existing analysisrun os-demo-web-v2020-04-21-391b-5bc768f98-48.1 (phase: Successful, equal: false, controllerUIDEqual: true)" namespace=os-demo-jkost rollout=os-demo-web-v2020-04-21-391b
  | time="2020-05-13T19:45:53Z" level=info msg="Encountered collision of existing analysisrun os-demo-web-v2020-04-21-391b-5bc768f98-48.2 (phase: Running, equal: false, controllerUIDEqual: true)" namespace=os-demo-jkost rollout=os-demo-web-v2020-04-21-391b

```

I can run a new sync and it will succeed but this leaves my rollout in a progressing state and never tears the old replicas down:
```
Name:            os-demo-web-v2020-04-21-391b
Namespace:       os-demo-jkost
Status:          ◌ Progressing
Strategy:        BlueGreen
Images:          <image_name> (active)
Replicas:
  Desired:       1
  Current:       2
  Updated:       1
  Ready:         2
  Available:     1

NAME                                                                KIND         STATUS         AGE    INFO
⟳ os-demo-web-v2020-04-21-391b                                      Rollout      ◌ Progressing  7d4h   
├──# revision:48                                                                                       
│  ├──⧉ os-demo-web-v2020-04-21-391b-5bc768f98                      ReplicaSet   ✔ Healthy      6m     active
│  │  └──□ os-demo-web-v2020-04-21-391b-5bc768f98-t5fk5             Pod          ✔ Running      6m     ready:1/1
│  ├──α os-demo-web-v2020-04-21-391b-5bc768f98-48.1                 AnalysisRun  ✔ Successful   5m23s  ✔ 2
│  │  └──⊞ 5663ed77-9552-11ea-ac85-001a4aa86510.datadog-validate.1  Job          ✔ Successful   5m23s  
│  ├──α os-demo-web-v2020-04-21-391b-5bc768f98-48                   AnalysisRun  ✔ Successful   5m23s  ✔ 2
│  │  └──⊞ 564cc5fb-9552-11ea-ac85-001a4aa86510.datadog-validate.1  Job          ✔ Successful   5m23s  
│  ├──α os-demo-web-v2020-04-21-391b-5bc768f98-48.2                 AnalysisRun  ✔ Successful   5m16s  ✔ 1
│  └──α os-demo-web-v2020-04-21-391b-5bc768f98-48.3                 AnalysisRun  ✔ Successful   5m15s  ✔ 1
├──# revision:47                                                                                       
│  ├──⧉ os-demo-web-v2020-04-21-391b-dd5768c8b                      ReplicaSet   ✔ Healthy      8m26s  delay:passed
│  │  └──□ os-demo-web-v2020-04-21-391b-dd5768c8b-smqmq             Pod          ✔ Running      8m26s  ready:1/1
│  ├──α os-demo-web-v2020-04-21-391b-dd5768c8b-47                   AnalysisRun  ✔ Successful   7m51s  ✔ 2
│  │  └──⊞ fe0bc431-9551-11ea-ac85-001a4aa86510.datadog-validate.1  Job          ✔ Successful   7m51s  
│  └──α os-demo-web-v2020-04-21-391b-dd5768c8b-47.1                 AnalysisRun  ✔ Successful   7m44s  ✔ 6
```

**Comment by johnkost** [comment]
Just noticed: the UI shows the rollout as finished with both replicasets still active

**Comment by jessesuen** [comment]
I believe this is addressed in v0.9.2. Please re-open if this is still an issue
